cmd/gomigen/main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+
+	"github.com/Makpoc/gomigen/internal/app"
+)
+
+func main() {
+	outputDir := flag.String(
+		"out",
+		"", // default is current dir
+		"optional path to output directory where the generated package with middleware files will be placed",
+	)
+	flag.Parse()
+
+	// TODO would it be better to make all args named?
+	//   e.g. `go run mw/.../gen -pkg . -iface Service [-out output_directory]`
+	if flag.NArg() < 2 {
+		log.Printf("Usage: %s [-out output_directory] <directory> <interface>", os.Args[0])
+		os.Exit(1)
+	}
+	var outDir string
+	if outputDir != nil {
```

Comment:
I think that this is guaranteed to not be nil, so your branch is unnecessary and you can just use `*outputDir` below.

Reply:
Changed